cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -119,34 +120,39 @@ impl LockedPackage {
         match self {
             LockedPackage::Catalog(pkg) => &pkg.install_id,
             LockedPackage::Flake(pkg) => &pkg.install_id,
+            LockedPackage::StorePath(pkg) => &pkg.install_id,
         }
     }
 
     pub(crate) fn system(&self) -> &System {
         match self {
             LockedPackage::Catalog(pkg) => &pkg.system,
             LockedPackage::Flake(pkg) => &pkg.locked_installable.system,
+            LockedPackage::StorePath(pkg) => &pkg.system,
         }
     }
 
     pub fn broken(&self) -> Option<bool> {
         match self {
             LockedPackage::Catalog(pkg) => pkg.broken,
             LockedPackage::Flake(pkg) => pkg.locked_installable.broken,
+            LockedPackage::StorePath(_) => None,
         }
     }
 
     pub fn unfree(&self) -> Option<bool> {
         match self {
             LockedPackage::Catalog(pkg) => pkg.unfree,
             LockedPackage::Flake(pkg) => pkg.locked_installable.unfree,
+            LockedPackage::StorePath(_) => None,
         }
     }
 
     pub fn derivation(&self) -> &str {
         match self {
             LockedPackage::Catalog(pkg) => &pkg.derivation,
             LockedPackage::Flake(pkg) => &pkg.locked_installable.derivation,
+            LockedPackage::StorePath(pkg) => pkg.store_path.as_str(),
```

Comment:
Since it's only used when upgrading with a catalog client and these packages never make it there could it be `None` to prevent confusion in the future? It would also satisfy the fact that store paths aren't upgradable if it were to ever leak through.

Reply:
true, made it an optional